core/unit_test/tools/TestEventCorrectness.hpp

```diff
@@ -450,13 +455,20 @@ TEST(kokkosp, parallel_scan_no_fence_view) {
         << "skipping since the OpenMPTarget backend has unexpected fences";
 #endif
 
+  // Execute the parallel_scan first without looking for fence events.
+  // Depending on the backend implementation and the order of tests,
+  // it might be that the first call to parallel_scan is reallocating scratch
```

Comment:
Ohh ... isn't this a bit sketchy, this dependency on backend and test order?  It sounds like the main problem is you're dealing with a fence that will be a natural consequence of deallocation, but that you do not necessarily want to count that particular fence ...?  If that assessment is correct, maybe we could rethink what we want to test (fence count?) and how we want to test it.

Reply:
The purpose is to check for undesired fences and we have to define what "undesired" means. Even if we are saying, "parallel_scan without result parameter doesn't fence" it is realistically very difficult not to allow for, say, internal allocations to be adjusted. It's debatable if we want to report a corresponding fence to our `Tools` interface but it exists regardless. The best we can test is "if all internals are set up, parallel_scan without result parameter doesn't fence". There are different ways to achieve that but the easiest seems to just run once without recording.
Of course, we could also opt for ignoring deallocation events which is what #6836 does but that seems even more fragile since it relies on the string used for the fencing when deallocating to be the same for every backend. Also, we shouldn't deallocate for repeated API calls anyway.